src/server_manager/web_app/gcp_account.ts

```diff
@@ -252,28 +263,41 @@ export class GcpAccount implements gcp.Account {
     };
     const createInstanceOperation =
         await this.apiClient.createInstance(zoneLocator, createInstanceData);
-    if (createInstanceOperation.error?.errors) {
-      // TODO: Throw error.
+    const errors = createInstanceOperation.error?.errors;
+    if (errors) {
+      throw new ServerInstallFailedError(`Instance creation failed: ${errors}`);
     }
 
-    const instanceLocator = {instanceId: createInstanceOperation.targetId, ...zoneLocator};
-    const instance = await this.apiClient.getInstance(instanceLocator);
+    const instanceId = createInstanceOperation.targetId;
+    const instanceCreation = this.apiClient.computeEngineOperationZoneWait(zoneLocator, createInstanceOperation.name).then(async () => {
+      const instanceLocator = {instanceId, ...zoneLocator};
+      try {
+        await this.promoteEphemeralIp(instanceLocator);
+      } catch (e) {
+        await this.apiClient.deleteInstance(instanceLocator);
+        throw e;
+      }
+    });
+
+    return {instanceId, instanceCreation};
+  }
 
+  private async promoteEphemeralIp(instanceLocator: gcp_api.InstanceLocator): Promise<void> {
+    const instance = await this.apiClient.getInstance(instanceLocator);
     // Promote ephemeral IP to static IP
     const ipAddress = instance.networkInterfaces[0].accessConfigs[0].natIP;
     const createStaticIpData = {
       name: instance.name,
       description: instance.description,
       address: ipAddress,
     };
-    const regionId = new gcp.Zone(zoneLocator.zoneId).regionId;
+    const regionId = new gcp.Zone(instanceLocator.zoneId).regionId;
     const createStaticIpOperation = await this.apiClient.createStaticIp(
-        {regionId, ...zoneLocator}, createStaticIpData);
-    if (createStaticIpOperation.error?.errors) {
-      // TODO: Delete VM instance. Throw error.
+        {regionId, ...instanceLocator}, createStaticIpData);
```

Comment:
Please create a region locator without the instance fields.

Reply:
Done